app/src/server/api/routers/billing.router.ts

```diff
@@ -0,0 +1,147 @@
+import { sql } from "kysely";
+import { z } from "zod";
+import { createTRPCRouter, protectedProcedure } from "~/server/api/trpc";
+import { kysely, prisma } from "~/server/db";
+import { v4 as uuidv4 } from "uuid";
+import { requireCanViewProject } from "~/utils/accessControl";
+import { toUTC } from "~/utils/dayjs";
+import { success } from "~/utils/errorHandling/standardResponses";
+import { getStats } from "./usage.router";
+
+export const billingRouter = createTRPCRouter({
+  invoices: protectedProcedure
+    .input(
+      z.object({
+        projectId: z.string(),
+      }),
+    )
+    .query(async ({ input, ctx }) => {
+      await requireCanViewProject(input.projectId, ctx);
+
+      const invoices = await kysely
+        .selectFrom("Invoice as i")
+        .where("i.projectId", "=", input.projectId)
+        .orderBy("i.createdAt", "desc")
+        .selectAll()
+        .execute();
+
+      return { invoices };
+    }),
+  invoice: protectedProcedure
+    .input(
+      z.object({
+        invoiceId: z.string(),
+      }),
+    )
+    .query(async ({ input, ctx }) => {
+      const invoice = await prisma.invoice.findUniqueOrThrow({
```

Comment:
Instead of `findUniqueOrThrow` (which I admit I've used more than I should), I would make a `findUnique` call here, and explicitly throw a 404 error if the invoiceId is not found. I believe findUniqueOrThrow results in a 500 error being thrown if the record is not found.